src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5285,6 +5285,48 @@ protected sealed override ISymbol GetEnclosingSymbolCore(int position, Cancellat
             return this.GetEnclosingSymbol(position, cancellationToken);
         }
 
+        private protected sealed override IImportScope GetImportScopeCore(int position, CancellationToken cancellationToken)
+        {
+            position = CheckAndAdjustPosition(position);
+            var binder = GetEnclosingBinder(position);
+            return ConvertToImportScope(binder?.ImportChain);
+
+            static IImportScope ConvertToImportScope(ImportChain chain)
+            {
+                // Skip by any empty items in the chain.
+                while (chain != null && chain.Imports.IsEmpty)
+                    chain = chain.ParentOpt;
+
+                // If we reached the end, there's nothing to return
+                if (chain == null)
+                    return null;
+
+                var imports = chain.Imports;
+                Debug.Assert(!imports.IsEmpty);
+
+                // Try to create a node corresponding to the imports of the next higher binder scope. Then create the
+                // node corresponding to this set of imports and chain it to that.
+                return new SimpleImportScope(
+                    ConvertToImportScope(chain.ParentOpt),
+                    ConvertAliases(imports),
+                    imports.ExternAliases.SelectAsArray(static e => e.Alias.GetPublicSymbol()),
+                    imports.Usings.SelectAsArray(static n => n.NamespaceOrType.GetPublicSymbol()),
+                    XmlNamespaces: ImmutableArray<string>.Empty);
+            }
+
+            static ImmutableArray<IAliasSymbol> ConvertAliases(Imports imports)
+            {
+                if (imports.UsingAliases.IsEmpty)
+                    return ImmutableArray<IAliasSymbol>.Empty;
+
+                var aliases = ArrayBuilder<IAliasSymbol>.GetInstance(imports.UsingAliases.Count);
+                foreach (var kvp in imports.UsingAliases)
+                    aliases.Add(kvp.Value.Alias.GetPublicSymbol());
+
+                return aliases.ToImmutableAndFree();
```

Comment:
>aliases.ToImmutableAndFree() [](http://example.com/codeflow?start=23&length=28)

It looks like the order of symbols is non-deterministic. Perhaps we should sort by name before producing the final array.